crypto/evp/evp_rand.c

```diff
@@ -194,6 +194,7 @@ static void *evp_rand_from_dispatch(int name_id,
             if (rand->get_ctx_params != NULL)
                 break;
             rand->get_ctx_params = OSSL_FUNC_rand_get_ctx_params(fns);
+            fnctxcnt++;
```

Comment:
Are you fixing an unrelated error here en passant?

Reply:
This guarantees that `get_context_params` is non-NULL.  The entire EVP_RAND breaks (well returns errors for just about everything) if it is NULL -- can't get _status_ or _strength_ otherwise.